build-logic/buildquality/src/main/kotlin/gradlebuild.configure-ci-artifacts.gradle.kts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import gradlebuild.basics.BuildEnvironment
+import gradlebuild.classycle.tasks.Classycle
+import gradlebuild.cleanup.tasks.KillLeakingJavaProcesses
+import gradlebuild.docs.FindBrokenInternalLinks
+import gradlebuild.integrationtests.tasks.DistributionTest
+import gradlebuild.performance.tasks.PerformanceTest
+import gradlebuild.testcleanup.extension.TestFilesCleanupBuildServiceRootExtension
+import me.champeau.gradle.japicmp.JapicmpTask
+import org.jlleitschuh.gradle.ktlint.tasks.GenerateReportsTask
+
+if (BuildEnvironment.isCiServer && project.name != "gradle-kotlin-dsl-accessors") {
+    val globalExtension = rootProject.extensions.getByType<TestFilesCleanupBuildServiceRootExtension>()
+    project.gradle.taskGraph.whenReady {
+        val allTasks = this@whenReady.allTasks
+        globalExtension.cleanupRunnerStep.set(allTasks.filterIsInstance<KillLeakingJavaProcesses>().isNotEmpty())
+        val taskPathToReports = allTasks.associate { it.path to it.customReports() + it.attachedReportLocations() }.filter { it.value.isNotEmpty() }
+        globalExtension.taskPathToReports.set(taskPathToReports)
```

Comment:
If I understand correctly this needs to run before the root plugin reads it. Is all this lazy enough for this to work? I suppose our test coverage would show that this works, right?

Reply:
Our test coverage will cover it, yes, that's why I create a new `DefaultProvider` above.